pkg/test/framework/components/echo/util/traffic/result.go

```diff
@@ -49,12 +50,25 @@ func (r *Result) add(resp client.ParsedResponses, err error) {
 
 	r.TotalRequests += count
 	if err != nil {
-		r.Error = multierror.Append(r.Error, err)
+		r.Error = multierror.Append(r.Error, fmt.Errorf("request %d: %v", r.TotalRequests, err))
 	} else {
 		r.SuccessfulRequests += count
 	}
 }
 
 func (r Result) PercentSuccess() float64 {
-	return (float64(r.SuccessfulRequests) / float64(r.TotalRequests)) * 100.0
+	return float64(r.SuccessfulRequests) / float64(r.TotalRequests)
+}
+
+// CheckSuccessRate asserts that a minimum success threshold was met.
+func (r Result) CheckSuccessRate(t test.Failer, minimumPercent float64) {
```

Comment:
nit: name doesn't imply this fails test (like how tf `Check...` return error)

Reply:
+1 you could just have 2 versions of the method. `CheckSuccessRate` that returns `error` and then `CheckSuccessRateOrFail` that calls `CheckSuccessRate` and fails a `Failer` if there's an error.

Reply:
We've been following this pattern pretty consistently throughout the test framework. There have been occasions where it's been useful. But I'm fine with leaving it as-is for now. We can always refactor later if it ends up being useful.